am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementServiceBenchmarks.java

```diff
@@ -0,0 +1,118 @@
+            new AccessManagementService(DataSourceBuilder
+                .create()
+                .username("amuser@am-lib-test-aat")
+                .password("upF#99gf7RAZ?77H")
+                .url("jdbc:postgresql://am-lib-test-aat.postgres.database.azure.com:5432/am?user=amuser@am-lib-test-aat&password=upF#99gf7RAZ?77H&sslmode=require")
+                .build()
+            );
+
+        //need to define role constant and add to AAT (for role based)
+
+        //define resource constant
+        Resource resource = Resource.builder()
+            .resourceId(resourceId)
+            .type(ResourceDefinition.builder()
+                .serviceName("Service 1")
+                .resourceType("Resource Type 1")
+                .resourceName("resource")
+                .build())
+            .resourceJson(inputJson)
+            .build();
+
+        @Setup
+        public void doSetup() {
+            //create map of explicit access JsonPointer -> Permission
+            Map<JsonPointer, Set<Permission>> attributePermissions =
+                ImmutableMap.<JsonPointer, Set<Permission>>builder()
+                    .put(JsonPointer.valueOf(""), ImmutableSet.of(READ, CREATE))
+                    .put(JsonPointer.valueOf("/orders"), ImmutableSet.of(CREATE))
+                    .put(JsonPointer.valueOf("/orders/orderType"), ImmutableSet.of(READ))
+                    .put(JsonPointer.valueOf("/orders/emergencyProtectionOrderDirections"), ImmutableSet.of(DELETE))
+                    .put(JsonPointer.valueOf("/caseName"), ImmutableSet.of(UPDATE, CREATE, READ, DELETE))
+                    .put(JsonPointer.valueOf("/children"), ImmutableSet.of(READ))
+                    .put(JsonPointer.valueOf("/children/firstChild"), ImmutableSet.of(UPDATE))
+                    .put(JsonPointer.valueOf("/children/adoption"), ImmutableSet.of(READ, CREATE))
+                    .put(JsonPointer.valueOf("/applicant"), ImmutableSet.of(READ, CREATE))
+                    .put(JsonPointer.valueOf("/solicitor"), ImmutableSet.of(UPDATE, DELETE, CREATE))
+                    .build();
+
+            //create explicit access grant
+            service.grantExplicitResourceAccess(ExplicitAccessGrant.builder()
+                .resourceId(resourceId)
+                .accessorIds(ImmutableSet.of(accessorId))
+                .accessType("EXPLICIT")
+                .serviceName("Service 1")
+                .resourceType("Resource Type 1")
+                .resourceName("resource")
+                .attributePermissions(attributePermissions)
+                .securityClassification(SecurityClassification.PUBLIC)
+                .build()
+            );
+        }
+    }
+
+    @Benchmark
+    public void baseline() {
+        //NO-OP
+    }
+
+    @Benchmark
+    public void filterResourceBenchmark(RecordState state) {
+        state.service.filterResource(state.accessorId, ImmutableSet.of("DOES_NOT_EXIST"), state.resource);
```

Comment:
This executes explicit access only, hoverer we would probably like to mix explicit and role based or have two benchmark in worse case scenario.